minidump-processor/src/processor.rs

```diff
@@ -468,17 +474,18 @@ where
     // would care about. So just providing an iterator and letting minidump-processor
     // pull out the things it cares about is simple and effective.
 
-    let mut cpu_microcode_version = None;
-    for (key, val) in linux_cpu_info.iter() {
-        if key.as_bytes() == b"microcode" {
-            cpu_microcode_version = val
-                .to_str()
-                .ok()
-                .and_then(|val| val.strip_prefix("0x"))
-                .and_then(|val| u64::from_str_radix(val, 16).ok());
-            break;
-        }
-    }
+    let cpu_microcode_version = linux_cpu_info
+        .iter()
+        .find_map(|(key, val)| {
+            if key.as_bytes() == b"microcode" {
+                val.to_str().ok()
+            } else {
+                None
+            }
+        })
+        .or_else(|| evil.cpu_microcode_version.as_ref().map(|s| s.as_str()))
+        .and_then(|val| val.strip_prefix("0x"))
+        .and_then(|val| u64::from_str_radix(val, 16).ok());
```

Comment:
Could you split this out in a helper function for better readability?

Reply:
To clarify, do you want the conversion from hex split out, or the entire block which determines the microcode version?

Reply:
The entire block, something like `get_microcode_version(linux_cpu_info, evil)`.

Reply:
Given the use of local state and unlikelihood of reuse, would it be more appropriate to add comments rather than adding a helper function if readability is an issue? Otherwise you could make the same statement (as far as adding helper functions) for each of the following values that are extracted from streams (`linux_standard_base`, `threads`, etc).

Otherwise, I think `let cpu_microcode_version = get_microcode_version(...)` is redundant in readability (the `let cpu_microcode_version = ...` already makes the functional behavior of the right side of the `=` fairly clear). However, I respect and defer to your final judgement, since you are the code owner :smile:

Reply:
The name was just a suggestion, you could use something more expressive like `microcode_from_cpuinfo_or_evil_json()` or something else that feels appropriate. This isn't because of reuse but to start shrinking this function. We were a bit in a rush when we wrote this so we've ended up with several-hundred-lines-long functions that could use some structure to make it clearer what's going on to someone who reads them the first time.

Reply:
Sure, that sounds like a worthy cause to me!

Reply:
Do you mind if I refactor the code to shrink the function further? Or would it be preferable to do your requested change here and save that for a separate PR?

Reply:
Make it in a separate PR, it's fine if it's just for refactoring, I'd be very happy about that.